### ISSUE TITLE
Bug in get_submap

### ISSUE THREAD

**Issue by marcoct** [probcomp/Gen.jl#74]

**Comment by marcoct** [comment]
The following fails:
```julia
@gen function foo()
    @trace(bernoulli(0.5), :x => 1)
    @trace(bernoulli(0.5), :x => 2)
end
trace = simulate(foo, ())
println(get_choices(trace))
@assert !isempty(get_submap(get_choices(trace), :x))
```

**Comment by marcoct** [comment]
Fixed in 44a7bcba786686b9352f1fd3b1e7e87a3513b035
